src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -158,10 +170,14 @@ inline void rjson_serialize(
     ::json::rjson_serialize(w, key.version);
     w.Key("magic");
     ::json::rjson_serialize(w, key.magic);
-    w.Key("seq");
-    ::json::rjson_serialize(w, key.seq);
-    w.Key("node");
-    ::json::rjson_serialize(w, key.node);
+    if (key.seq.has_value()) {
+        w.Key("seq");
+        ::json::rjson_serialize(w, key.seq);
+    }
```

Comment:
I'm curious why we would prefer omission over `seq: null` nullness.

Reply:
> I'm curious why we would prefer omission over `seq: null` nullness.

So that the record key matches for 3rdparty compatibility - let's say a config or schema is hard deleted, we need to match the key (without `seq` or `node`), and provide no record value, and let compaction do its thing.